unlock-app/src/components/interface/locks/Settings/forms/CreditCardForm.tsx

```diff
@@ -151,7 +152,13 @@ const ConnectStripe = ({
   }
 
   const { isLoading: isLoadingCheckGrantedStatus, data: isGranted } = useQuery(
-    ['checkIsKeyGranter', lockAddress, network, keyGranter],
+    [
+      'checkIsKeyGranter',
+      lockAddress,
+      network,
+      keyGranter,
+      grantKeyGrantorRoleMutation.isSuccess,
```

Comment:
why not use isEnabled here? 🤔

Reply:
need to check `isGranted` status when i access the page or when `grantKeyGrantorRoleMutation` is done, 
with `enabled` will get stuck on page reload 🤔

Reply:
No, that is fine. I'm saying you can use this is in enabled option right, `enabled: grantKeyGrantorRoleMutation.isSuccess`

Reply:
This is going to be buggy. You should directly call refetch when mutation is called to get new data.